### ISSUE TITLE
Winbar support

### ISSUE THREAD

**Issue by wookayin** [dstein64/nvim-scrollview#72]
Neovim (0.8.0+ nightly) recently added a new feature 'winbar' (https://github.com/neovim/neovim/pull/18562). When winbar is enabled, a single line of per-window statusline will appear at the top of a window, which decreases the effective number of editor lines available for the window by one.

<img width="961" alt="" src="https://user-images.githubusercontent.com/1009873/172267140-8bc59ce1-adc5-4939-b36c-575239459e93.png">

As a result, scrollbars will appear on the statusline or span the window boundary if there is a split. Please see the screenshot above.

**Comment by wookayin** [comment]
One possible fix for this would be replace all the use of `api.nvim_win_get_height(winid)` with something like this (the idea is similar to https://github.com/j-hui/fidget.nvim/pull/82 but the function should work for non-current-windows as well):
```lua
-- Returns the number of rows in the window, excluding winbar (if any).
local get_win_height = function(winnr)
  local height = vim.api.nvim_win_get_height(winnr)
  if vim.fn.exists('+winbar') > 0 > 0 then
    -- Resolve winbar option, from local opt and then from global opt.
    local winbar = api.nvim_win_get_option(winnr, 'winbar')
    if winbar == "" then
      winbar = vim.opt.winbar:get()
    end
    if winbar ~= "" then
      -- winbar is used, so the effective height of window should be offset by 1.
      height = height - 1
    end
  end
  return height
end
```

This seems to be working in some cases but I haven't figured out a fully working solution, so I think it's better to be leaved to the original author @dstein64. Thank you for your consideration!

**Comment by dstein64** [comment]
Thanks for reporting the issue and proposing a fix @wookayin!

This is addressed in 9f2dacfb53274460e71f233c4257bd77e068f434.

**Comment by wookayin** [comment]
Thanks! Looks elegant and works perfectly. I've also learned `getwininfo` and `autocmd OptionSet` :)
